src/parser/ast.rs

```diff
@@ -118,26 +162,36 @@ impl BinModifier {
     }
 
     pub fn is_labels_joint(&self) -> bool {
-        matches!((self.card.labels(), &self.matching),
-                 (Some(labels), Some(matching)) if !matching.labels().is_disjoint(labels))
+        matches!(
+            (self.card.labels(), &self.matching),
+            (Some(labels), Some(matching)) if labels.is_joint(matching.labels())
+        )
     }
 
-    pub fn intersect_labels(&self) -> Option<Vec<&String>> {
+    pub fn intersect_labels(&self) -> Option<Vec<String>> {
         if let Some(labels) = self.card.labels() {
             if let Some(matching) = &self.matching {
-                return Some(matching.labels().intersection(labels).collect());
+                return Some(labels.intersect(matching.labels()).labels);
             }
         };
         None
     }
 
     pub fn is_matching_on(&self) -> bool {
-        matches!(&self.matching, Some(matching) if matching.is_on())
+        matches!(&self.matching, Some(matching) if matching.is_include())
     }
 
     pub fn is_matching_labels_not_empty(&self) -> bool {
         matches!(&self.matching, Some(matching) if !matching.labels().is_empty())
     }
+
+    pub fn bool_string(&self) -> &str {
```

Comment:
nitpick: It returns a str, not a string.
```suggestion
    pub fn bool_str(&self) -> &str {
```